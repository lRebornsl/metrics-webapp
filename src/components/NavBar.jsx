import settings_icon from "../assets/settings_icon.png";
import menu_icon from "../assets/menu_icon.png";

const NavBar = () => (
  <div className="px-4 py-2 text-center bg-yonder flex justify-between">
    <img src={menu_icon} alt="Icon Image"></img>
    <h1 className="text-white">MMO GAMES</h1>
    <img src={settings_icon} alt="Icon Image"></img>
  </div>
)

export default NavBar;