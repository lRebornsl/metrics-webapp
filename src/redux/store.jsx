import { configureStore } from '@reduxjs/toolkit';
import categoriesSlice from './Categories/categoriesSlice';

const store = configureStore({
  reducer: {
    categoriesSlice,
  },
});

export default store;