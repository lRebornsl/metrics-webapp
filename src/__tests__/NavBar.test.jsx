import { Provider } from "react-redux";
import { beforeEach, describe, expect, it } from "vitest";
import store from "../redux/store";
import NavBar from "../components/NavBar";
import { render, waitFor, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("NavBar tests", () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      </Provider>
    );
  });

  it("NavBar renders correctly", async () => {
    await waitFor(() => expect(NavBar).toMatchSnapshot());
  });
  
  it("Icons are rendered correctly", async () => {
    const menuIcon = screen.getByAltText("Menu");
    const settingsIcon = screen.getByAltText("Settings");
    expect(menuIcon).toMatchSnapshot();
    expect(settingsIcon).toMatchSnapshot();
  });
})