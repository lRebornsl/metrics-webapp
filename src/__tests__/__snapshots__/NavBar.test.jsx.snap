// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar tests > Icons are rendered correctly 1`] = `
<img
  alt="Menu"
  src="/src/assets/menu_icon.png"
/>
`;

exports[`NavBar tests > Icons are rendered correctly 2`] = `
<img
  alt="Settings"
  class="h-full"
  src="/src/assets/settings_icon.png"
/>
`;

exports[`NavBar tests > NavBar renders correctly 1`] = `[Function]`;
